wpiutil/src/main/native/include/wpi/mDNSResolver.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <wpi/span.h>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <utility>
+
+namespace wpi {
+class mDNSResolver {
+ public:
+  typedef void mDnsRevolveCompletion(
+      unsigned int ipv4Address, std::string_view serviceName,
+      std::string_view hostName,
+      wpi::span<const std::pair<std::string, std::string>> txt);
+  using mDnsRevolveCompletionFunc = std::function<mDnsRevolveCompletion>;
+
+  mDNSResolver(std::string_view serviceType, mDnsRevolveCompletionFunc onFound);
+  ~mDNSResolver() noexcept;
+
+  void Start();
+  void Stop();
+
+  struct Impl;
```

Comment:
Is there a reason why this isn't private as well?

Reply:
Because its being access from C callbacks, and I can't use a friend here because the callbacks use platform specific parameters that I don't want to polute.